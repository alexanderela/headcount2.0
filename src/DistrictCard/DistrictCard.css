.DistrictCard {
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: center;
	font-family: 'Open Sans', sans-serif;
	text-align: center;
	width: 15rem;
	height: 26rem;
	margin: 0 auto 1.5rem;
	box-shadow: 0px 1px 1px 2px #8d8686;
	border-radius: 10px;
}

.card-location {
	text-align: center;
	overflow-wrap: break-word;
	margin: 1rem 0.5rem 0.75rem;
}

.card-stats {
	margin: 0 auto 1.5rem;

}

.greater-than-point-5 {
	color: green;
	margin: 0.25rem auto;
}

.less-than-point-5 {
	color: red;
	margin: 0.25rem auto;

}

