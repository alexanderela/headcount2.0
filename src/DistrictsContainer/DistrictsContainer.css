.DistrictsContainer {
	display: grid;
	grid-template-columns: repeat(5, 1fr);
	margin: 0 5rem;
}

