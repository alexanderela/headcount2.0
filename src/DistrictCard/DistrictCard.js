import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';

const DistrictCard = ({ location, stats }) => {
	const statsKeys = Object.keys(stats)
	const schoolData = statsKeys.map((stat) => {
		
		return <p 
			className={(stats[stat] > 0.5) ? 'greater-than-point-5' : 'less-than-point-5'}>
			{stat}: {stats[stat]}</p>
	})
	console.log(schoolData)

	return (
		<div className='DistrictCard'>
			<h2 className='card-location'>{location}</h2>
			{schoolData}
		</div>
	)
}

DistrictCard.proptypes = {
	location: PropTypes.string.isRequired,
	stats: PropTypes.object.isRequired
}

export default DistrictCard;

			// className={(stats[stat] > 0.5) ? '.greater-than-point-5' : '.less-than-point-5'}>
			// {stat}: {stats[stat]}</p>