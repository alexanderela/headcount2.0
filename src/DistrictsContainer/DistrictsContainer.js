import React from 'react';
import './DistrictsContainer.css';
import DistrictCard from '../DistrictCard/DistrictCard.js';

const DistrictsContainer = ({districts}) => {
	// const districtCards = districts.map((district) => {
	// 	return <DistrictCard {...district}/>
	// })

	return (
		<div className='DistrictsContainer'>
			{/*{ districtCards }*/}
		</div>
	)
}

export default DistrictsContainer;