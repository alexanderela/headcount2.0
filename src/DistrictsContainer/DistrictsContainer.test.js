import React from 'react';
import ReactDOM from 'react-dom';
import DistrictsContainer from './DistrictsContainer.js';
import DistrictCard from '../DistrictCard/DistrictCard.js';
import { shallow } from 'enzyme';

// it('renders without crashing', () => {
// 	const div = document.createElement('div');
// 	ReactDOM.render(<DistrictsContainer />, div)
// })

describe('DistrictsContainer', () => {
	let wrapper;
	let mockDistricts;

	beforeEach(() => {
		mockDistricts = {'colorado': {}, 'academy 20': {}} 
		wrapper = shallow(<DistrictsContainer districts={mockDistricts} />)
	})

	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it.skip('renders each district', () => {
		expect(wrapper.find(DistrictCard).length).toEqual(3)
	})
})
