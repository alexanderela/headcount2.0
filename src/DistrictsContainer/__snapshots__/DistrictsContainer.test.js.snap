// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictsContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictsContainer
    districts={
      Object {
        "academy 20": Object {},
        "agate 300": Object {},
        "colorado": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DistrictCard
          schoolName="colorado"
        />,
        <DistrictCard
          schoolName="academy 20"
        />,
        <DistrictCard
          schoolName="agate 300"
        />,
      ],
      "className": "DistrictsContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "colorado",
        "nodeType": "function",
        "props": Object {
          "schoolName": "colorado",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "academy 20",
        "nodeType": "function",
        "props": Object {
          "schoolName": "academy 20",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "agate 300",
        "nodeType": "function",
        "props": Object {
          "schoolName": "agate 300",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DistrictCard
            schoolName="colorado"
          />,
          <DistrictCard
            schoolName="academy 20"
          />,
          <DistrictCard
            schoolName="agate 300"
          />,
        ],
        "className": "DistrictsContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "colorado",
          "nodeType": "function",
          "props": Object {
            "schoolName": "colorado",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "academy 20",
          "nodeType": "function",
          "props": Object {
            "schoolName": "academy 20",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "agate 300",
          "nodeType": "function",
          "props": Object {
            "schoolName": "agate 300",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
